### PR TITLE
Wavefield injection based on wavefield discontinuity

### PR DESCRIPTION
**Pull request by tianshi-liu**
An implementation of wavefield injection based on wavefield discontinuity which naturally accommodates PML/Stacey. An small example for coupling with FK is provided in EXAMPLES/applications/wavefield_discontinuity. More examples to be provided in future commits. Currently only forward simulation is implemented, adjoint simulation and kernel calculation will come soon. HDF5 and GPU supports will be implemented in the near future.

### CODE REVIEW

File: src/specfem3D/compute_forces_viscoelastic.F90 [hunk 1]
```diff
@@ -312,6 +317,15 @@ subroutine compute_forces_viscoelastic(iphase,deltat, &
           enddo
         enddo
       enddo
+      !! solving wavefield discontinuity problem with non-split-node scheme
+      !! add back displacement discontinuity for inner elements of
+      !! the discontinuity interface
+      !! note that this is called in adjoint simulation
+      if (IS_WAVEFIELD_DISCONTINUITY .and. &
```
Comment (danielpeter) [inline src/specfem3D/compute_forces_viscoelastic.F90:325]: this breaks the OpenMP version. 

please add the `IS_WAVEFIELD_DISCONTINUITY` variable to the `$OMP SHARED(..)` section to fix it.

File: DATA/CMTSOLUTION
Comment (danielpeter) [inline DATA/CMTSOLUTION:1]: that's a bit unfortunate to delete and then re-insert this file. git won't be able to track changes in this file like this. 

if still possible, can you try to revert back to the initial file? note that the files in folder SPECFEM3D/DATA/ are all symbolic links to the files in the main "default" example `EXAMPLES/applications/homogeneous_halfspace/DATA/`.

File: DATA/Par_file
Comment (danielpeter) [inline DATA/Par_file:1]: same with this file

File: DATA/STATIONS
Comment (danielpeter) [inline DATA/STATIONS:1]: and this file.

**Comment by danielpeter** [comment]
i'll leave the PR open for now. please try to fix the above mentioned issues by adding the necessary commits to your devel version.

**Comment by tianshi-liu** [comment]
Hi Daniel, thanks for the comments! I have restored the symbolic links in the DATA/ directory, and I have added the IS_WAVEFIELD_DISCONTINUITY flag to the OMP directive. I hope it can pass the tests now.

**Comment by danielpeter** [comment]
great, looks all good - thanks for the quick fix!

**Comment by tianshi-liu** [comment]
Thanks for helping me with my first pull request!

**Pull request merged**